packages/components/src/Actions/ActionFile/ActionFile.component.js

```diff
@@ -61,82 +61,95 @@ function getButtonContent(props) {
  * subsequently the field get emptied.
  * @param {Object} props
  */
-function ActionFile(props) {
-	const {
-		id,
-		inProgress,
-		disabled,
-		hideLabel,
-		label,
-		name,
-		onChange,
-		tooltipPlacement,
-		tooltip,
-		tooltipLabel,
-		available,
-	} = props;
-	if (!available) {
-		return null;
+class ActionFile extends React.Component {
+	static displayName = 'ActionFile';
+
+	static propTypes = {
+		...getIcon.propTypes,
+		id: PropTypes.string,
+		bsStyle: PropTypes.string,
+		disabled: PropTypes.bool,
+		hideLabel: PropTypes.bool,
+		iconPosition: PropTypes.oneOf([LEFT, RIGHT]),
+		label: PropTypes.string.isRequired,
+		link: PropTypes.bool,
+		model: PropTypes.object,
+		name: PropTypes.string,
+		onChange: PropTypes.func.isRequired,
+		tooltipPlacement: OverlayTrigger.propTypes.placement,
+		tooltip: PropTypes.bool,
+		tooltipLabel: PropTypes.string,
+	};
+
+	static defaultProps = {
+		available: true,
+		bsStyle: 'default',
+		tooltipPlacement: 'top',
+		inProgress: false,
+		disabled: false,
+	};
+
+	constructor(props) {
+		super(props);
+		this.handleChange = this.handleChange.bind(this);
 	}
-	const localId = id || uuid.v4();
-	const buttonContent = getButtonContent(props);
-	const labelClasses = classNames('btn', theme['btn-file'], (disabled || inProgress) && 'disabled');
-	const localOnChange = event => {
+	handleChange(event) {
 		event.preventDefault();
-		onChange(event, event.target.files[0]);
-		// clear input value
-		event.target.value = null; /* eslint no-param-reassign: ["error", { "props": false }] */
-	};
-	const btn = (
-		<span>
-			<input
-				onChange={localOnChange}
-				type="file"
-				name={name}
-				id={localId}
-				disabled={inProgress || disabled}
-				className={theme['action-file-label']}
-			/>
-			<label htmlFor={localId} className={labelClasses}>
-				{buttonContent}
-			</label>
-		</span>
-	);
-	if (hideLabel || tooltip || tooltipLabel) {
-		return (
-			<TooltipTrigger label={tooltipLabel || label} tooltipPlacement={tooltipPlacement}>
-				{btn}
-			</TooltipTrigger>
+		this.props.onChange(event, event.target.files[0]);
+		this.form.reset();
+	}
+
+	render() {
+		const {
+			id,
+			name,
+			available,
+			disabled,
+			inProgress,
+			hideLabel,
+			tooltip,
+			tooltipLabel,
+			label,
+			tooltipPlacement,
+		} = this.props;
+		if (!available) {
+			return null;
+		}
+		const localId = id || uuid.v4();
+		const buttonContent = getButtonContent(this.props);
+		const labelClasses = classNames(
+			'btn',
+			theme['btn-file'],
+			(disabled || inProgress) && 'disabled',
+		);
+		const btn = (
+			<form
```

Comment:
I know that @frassinier prefer that all input need to nest in a form excect that if the action is integrate in a form, this form is ignored.

try this https://jsfiddle.net/4j6dravd/ and look the DOM. The second form nested in a form is removed:

![nested form](https://user-images.githubusercontent.com/20772261/32937649-ad7362d0-cb79-11e7-994b-44c937d46695.png)